sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveUdfSuite.scala

```diff
@@ -133,6 +134,41 @@ class HiveUdfSuite extends QueryTest {
     TestHive.reset()
   }
 
+  test("UDFToListString") {
+    val testData = TestHive.sparkContext.parallelize(StringCaseClass("") :: Nil).toDF()
+    testData.registerTempTable("inputTable")
+
+    /**
+     * Converts $"..." into UTF8String(...).
+     * TODO: Catalyst (e.g., CatalystTypeConverters) cannot convert UTF8String
+     * into String correctly because of JVM type erasure.
+     */
+    implicit class StringToUtf8(val sc: StringContext) {
+      def u(args: Any*): UTF8String = UTF8String(sc.s(args :_*))
+    }
+
+    sql(s"CREATE TEMPORARY FUNCTION testUDFToListString AS '${classOf[UDFToListString].getName}'")
+    checkAnswer(
+      sql("SELECT testUDFToListString(s) FROM inputTable"),
+      Seq(Row(u"data1" :: u"data2" :: u"data3" :: Nil)))
```

Comment:
Yes, some libraries already depend on this type of UDFs:
https://github.com/myui/hivemall/blob/master/src/main/java/hivemall/ftvec/AddBiasUDF.java#L37

Any idea to avoid this leaking?